app.js

```diff
@@ -1,10 +1,39 @@
+/*jslint node: true */  
 'use strict';
 
 var ari = require('ari-client');
 var util = require('util');
 var sla = require('./lib/sla.js');
+var Q = require('q');
 
-ari.connect('http://127.0.0.1:8088', 'user', 'pass', clientLoaded);
+var connect = Q.denodeify(ari.connect);
 
-function clientLoaded (err, client){
-    console.log(sla());}
+connect('http://127.0.0.1:8088', 'user', 'pass')
+  .then(function (client) {
+    clientLoaded(client);
+  })
+  .catch(errHandler);
```

Comment:
You should call done here so the error you throw in errHandler can propagate up the stack.